### PR TITLE
Adding XMLsec tests

### PR DESCRIPTION
**Pull request by nasastry**
security/xmlsec-tests.py adding XMLSec unit tests

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
JOB ID     : 8cb6308b4c56583b881340b98d1234ef1f491c8a
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-29T11.11-8cb6308/job.log
 (1/1) xmlsec-tests.py:XMLSec.test: FAIL: 9 test(s) failed, please refer to the log (41.58 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-29T11.11-8cb6308/results.html
JOB TIME   : 42.27 s
```

**Comment by nasastry** [comment]
RHEL:
```
Fetching asset from xmlsec-tests.py:XMLSec.test
JOB ID     : ab7b8a70e4c484b5cb4e6ae7a303636c9200d174
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-02T12.48-ab7b8a7/job.log
 (1/1) xmlsec-tests.py:XMLSec.test: FAIL: 9 test(s) failed, please refer to the log (45.58 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-02T12.48-ab7b8a7/results.html
JOB TIME   : 46.60 s
```

SLES:
```
Fetching asset from xmlsec-tests.py:XMLSec.test
JOB ID     : 99b1ab9239760d29044abbb55d5df12857881572
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-06T09.00-99b1ab9/job.log
 (1/1) xmlsec-tests.py:XMLSec.test: PASS (120.85 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-06T09.00-99b1ab9/results.html
JOB TIME   : 122.52 s
```

File: security/xmlsec-tests.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    xmlsec-testsuite
+    :avocado: tags=security,testsuite
+    """
+    def setUp(self):
+        '''
+        Install the basic packages to support xmlsec
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        deps = ['gcc', 'make', 'autoconf', 'libxml2-devel', 'libtool-ltdl-devel']
```
Comment (harish-24) [inline security/xmlsec-tests.py:34]: Looks good. As we are adding packages to generic list make sure it is handled for Debian based distros.

**Comment by nasastry** [comment]
@harish-24 added Ubuntu part also

Ubuntu:
```
JOB ID     : 16f4207a12615007f842e8d0baccf4db4199c6b2
JOB LOG    : /root/avocado/job-results/job-2020-10-05T09.44-16f4207/job.log
 (1/1) xmlsec-tests.py:XMLSec.test: PASS (57.11 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 57.47 s
```

RHEL:
```
Fetching asset from xmlsec-tests.py:XMLSec.test
JOB ID     : 0007c08943de97362cbeee05ba29ced2b36378c0
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-05T10.46-0007c08/job.log
 (1/1) xmlsec-tests.py:XMLSec.test: FAIL: 9 test(s) failed, please refer to the log (47.61 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-05T10.46-0007c08/results.html
JOB TIME   : 47.68 s
```

SLES:
```
Fetching asset from xmlsec-tests.py:XMLSec.test
JOB ID     : 3042507545f9367ecb7b1c18b9fca5ad00fc38a7
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-09T06.59-3042507/job.log
 (1/1) xmlsec-tests.py:XMLSec.test: PASS (105.73 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-09T06.59-3042507/results.html
JOB TIME   : 106.75 s
```

**Review by harish-24 — APPROVED** [review]
LGTM

**Pull request merged**